contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,135 @@
+    using SafeERC20 for IERC20;
+
+    bool public receivedSettlementPrice;
+
+    bool public enableEarlyExpiration; // If set, the LSP contract can request to be settled early by calling the OO.
+    uint256 public expirationTimestamp;
+    string public pairName;
+    uint256 public collateralPerPair; // Amount of collateral a pair of tokens is always redeemable for.
+
+    // Number between 0 and 1e18 to allocate collateral between long & short tokens at redemption. 0 entitles each short
+    // to collateralPerPair and each long to 0. 1e18 makes each long worth collateralPerPair and short 0.
+
+    // Number between 0 and 1e18 to allocate collateral between long & short tokens at redemption. 0 entitles each short
+    // to collateralPerPair and each long to 0. 1e18 makes each long worth collateralPerPair and short 0.
+    uint256 public expiryPercentLong;
+    bytes32 public priceIdentifier;
+
+    // Price returned from the Optimistic oracle at settlement time.
+    int256 public expiryPrice;
+
+    // External contract interfaces.
+    IERC20 public collateralToken;
+    ExpandedIERC20 public longToken;
+    ExpandedIERC20 public shortToken;
+    FinderInterface public finder;
+    LongShortPairFinancialProductLibrary public financialProductLibrary;
+
+    // Optimistic oracle customization parameters.
+    bytes public customAncillaryData;
+    uint256 public proposerReward;
+    uint256 public optimisticOracleLivenessTime;
+    uint256 public optimisticOracleProposerBond;
+
+    // Define the contract's constructor parameters as a struct to enable more variables to be specified.
+    struct ConstructorParams {
+        string pairName; // Name of the long short pair contract.
+        uint256 expirationTimestamp; // Unix timestamp of when the contract will expire.
+        uint256 collateralPerPair; // How many units of collateral are required to mint one pair of synthetic tokens.
+        bytes32 priceIdentifier; // Price identifier, registered in the DVM for the long short pair.
+        // bool enableEarlyExpiration; // Enables the LSP contract to be settled early.
+        ExpandedIERC20 longToken; // Token used as long in the LSP. Mint and burn rights needed by this contract.
+        ExpandedIERC20 shortToken; // Token used as short in the LSP. Mint and burn rights needed by this contract.
+        IERC20 collateralToken; // Collateral token used to back LSP synthetics.
+        LongShortPairFinancialProductLibrary financialProductLibrary; // Contract providing settlement payout logic.
+        bytes customAncillaryData; // Custom ancillary data to be passed along with the price request to the OO.
+        uint256 proposerReward; // Optimistic oracle reward amount, pulled from the caller of the expire function.
+        uint256 optimisticOracleLivenessTime; // OO liveness time for price requests.
+        uint256 optimisticOracleProposerBond; // OO proposer bond for price requests.
```

Comment:
we could let the creator of the prediction market define these I think. the vision I have for this is that someone calls a method like `createPredictionMarket` and they define the `identifier` and `customAncillaryData`.

for `optimisticOracleLivenessTime` and `optimisticOracleProposerBond` we might as well default to some _reasnoble_ values.

remember the goal of this contract is to be as simple as posible and I think a part of this will be having less settings and less paramaters where posible!